File: backend/app/schemas/parser_errors.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, Any
from datetime import datetime

class ParserErrorUpdate(BaseModel):
    status: str
    resolution_notes: Optional[str] = None

class ParserErrorListItem(BaseModel):
    id: int
    import_batch_id: int
    source_page: int
    error_type: str
    severity: str
    raw_text: Optional[str]
    error_message: str
    college_code: Optional[str]
    course_code: Optional[str]
    status: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

class ParserErrorDetail(ParserErrorListItem):
    context: Optional[Any]

    model_config = ConfigDict(from_attributes=True)

class ParserErrorSummary(BaseModel):
    by_severity: dict
    by_type: dict
    by_batch: dict

File: frontend/src/api/parserErrors.ts
import apiClient from './client';
import { PaginatedResponse } from '../types/common';

export interface ParserError {
  id: number;
  import_batch_id: number;
  source_page?: number;
  error_type: string;
  severity: string;
  raw_text?: string | null;
  error_message: string;
  college_code?: string;
  course_code?: string;
  status: string;
  context?: any;
  created_at: string;
}

export const getParserErrors = async (params?: {
  page?: number;
  page_size?: number;
  import_batch_id?: number;
  severity?: string;
  error_type?: string;
  status?: string;
}) => {
  const { data } = await apiClient.get<PaginatedResponse<ParserError>>('/parser-errors', { params });
  return data;
};

export const getParserError = async (id: number) => {
  const { data } = await apiClient.get<ParserError>(`/parser-errors/${id}`);
  return data;
};

export const updateParserError = async (id: number, updateData: { status: string; resolution_notes?: string }) => {
  const { data } = await apiClient.put<ParserError>(`/parser-errors/${id}`, updateData);
  return data;
};

export const getParserErrorSummary = async () => {
  const { data } = await apiClient.get('/parser-errors/summary');
  return data;
};
